test/e2e/framework/service/jig.go

```diff
@@ -764,16 +764,26 @@ func testReachabilityOverClusterIP(clusterIP string, sp v1.ServicePort, execPod
 	return nil
 }
 
-func testReachabilityOverNodePorts(nodes *v1.NodeList, sp v1.ServicePort, pod *v1.Pod) error {
+func testReachabilityOverNodePorts(nodes *v1.NodeList, sp v1.ServicePort, pod *v1.Pod, clusterIP string) error {
 	internalAddrs := e2enode.CollectAddresses(nodes, v1.NodeInternalIP)
 	externalAddrs := e2enode.CollectAddresses(nodes, v1.NodeExternalIP)
+	isClusterIPV4 := net.ParseIP(clusterIP).To4() != nil
+
 	for _, internalAddr := range internalAddrs {
 		// If the node's internal address points to localhost, then we are not
 		// able to test the service reachability via that address
 		if isInvalidOrLocalhostAddress(internalAddr) {
 			framework.Logf("skipping testEndpointReachability() for internal adddress %s", internalAddr)
 			continue
 		}
+		isNodeInternalIPV4 := net.ParseIP(internalAddr).To4() != nil
+		// Check service reachability on the node internalIP which is same family
+		// as clusterIP
+		if isClusterIPV4 != isNodeInternalIPV4 {
+			framework.Logf("skipping testEndpointReachability() for internal adddress %s as it does not match clusterIP (%s) family", internalAddr, clusterIP)
+			continue
+		}
+
```

Comment:
lgtm
what about the underneath loop ? 
`	for _, externalAddr := range externalAddrs {`
is ok to left it as is?

Reply:
done!